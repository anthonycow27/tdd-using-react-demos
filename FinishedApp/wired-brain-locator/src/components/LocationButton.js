import React, { Component } from 'react';
import './LocationButton.css';

class LocationButton extends Component {
    render(){
        return(<button className="Location-button" onClick={this.props.changeMap}>{this.props.shop}</button>);
    }
}

export default LocationButton;