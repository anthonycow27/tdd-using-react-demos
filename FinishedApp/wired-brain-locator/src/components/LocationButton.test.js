import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LocationButton from './LocationButton';

const shops=["location1","location2"];
const changeMap = jest.fn();

describe('LocationButton', () => {

    it('renders without crashing', () => {
        const component = shallow(<LocationButton shop="Location1" changeMap = {changeMap} />);
        expect(component).toBeDefined();
    });

    it('matches the snapshot', () => {
        const component = shallow(<LocationButton shop="Location1" changeMap = {changeMap} />);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('calls the function on click', () => {
        const component = shallow(<LocationButton shop="Location1" changeMap = {changeMap} />);
        component.find('button').simulate('click');
        expect(changeMap.mock.calls.length).toBe(1);
    });
});