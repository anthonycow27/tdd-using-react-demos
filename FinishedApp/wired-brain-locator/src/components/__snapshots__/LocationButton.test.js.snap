// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationButton matches the snapshot 1`] = `
<button
  className="Location-button"
  onClick={[Function]}
  value="Location1"
>
  Location1
</button>
`;
